test/e2e/elasticache/util.py

```diff
@@ -0,0 +1,135 @@
+# Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may
+# not use this file except in compliance with the License. A copy of the
+# License is located at
+#
+#	 http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+"""ElastiCache-specific test utility functions
+"""
+
+import logging
+import boto3
+
+from time import sleep
+from typing import Tuple
+from common import k8s
+
+"""
+Waits for the specified condition in .status.conditions to reach the desired value.
+
+Precondition: resource must be consumed by the controller (i.e. have a .status field)
+
+This differs from some methods in the k8s.py module (e.g. wait_resource_synced) due to
+different representations of conditions across services. For example, that 
+method expects .status.conditions to be a dictionary; for ElastiCache it is
+a list.
+"""
+def wait_on_condition(reference: k8s.CustomResourceReference,
+                         condition_name: str,
+                         desired_condition_status: str,
+                         wait_periods: int = 2,
+                         period_length: int = 60) -> bool:
+    if not k8s.get_resource_exists(reference):
+        logging.error(f"Resource {reference} does not exist")
+        return False
+
+    for _ in range(wait_periods):
+        resource = k8s.get_resource(reference)
+
+        if 'conditions' not in resource['status']:
+            logging.error(f"Resource {reference} does not have a .status.conditions field.")
+            return False
+
+        desired_condition = None
+        for condition in resource['status']['conditions']:
+            if condition['type'] == condition_name:
+                desired_condition = condition
+
+        if not desired_condition:
+            logging.error(f"Resource {reference} does not have a condition of type {condition_name}.")
+            return False
+        else:
+            if desired_condition['status'] == desired_condition_status:
+                logging.info(f"Condition {condition_name} has status {desired_condition_status}, continuing...")
+                return True
+
+        sleep(period_length)
+
+    logging.error(f"Wait for condition {condition_name} to reach status {desired_condition_status} timed out")
+    return False
```

Comment:
Nice work on this @echen-98. Definitely something we should pull into https://github.com/aws-controllers-k8s/community/blob/main/test/e2e/common/k8s.py though!

Reply:
Thanks :) There's a little confusion here regarding the structure of conditions across services (see below), once that's resolved I can place it in there.

I was also thinking it would be nice to have a generalized wait function where you somehow specify the path of the field you want to on, although that can come in a future PR.